protocol/swap/manager.go

```diff
@@ -37,10 +35,12 @@ type manager struct {
 	past    map[types.Hash]*Info
 }
 
+var _ Manager = (*manager)(nil)
+
 // NewManager returns a new Manager that uses the given database.
 // It loads all ongoing swaps into memory on construction.
 // Completed swaps are not loaded into memory.
-func NewManager(db Database) (*manager, error) {
+func NewManager(db Database) (Manager, error) {
```

Comment:
We could move the `Manager` interface out of this package and capitalize the `manager` struct, but it seemed out of scope for upgrading the linter. This was to get rid of the complaint that a public function was returning a private type.

Reply:
I think we should do that, in another PR